src/drivers/roboclaw/roboclaw_params.c

```diff
@@ -0,0 +1,134 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013-2019 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file roboclaw_params.c
+ *
+ * Parameters defined by the Roboclaw driver.
+ *
+ * The Roboclaw will need to be configured to match these parameters. For information about configuring the
+ * Roboclaw, see http://downloads.ionmc.com/docs/roboclaw_user_manual.pdf
+ *
+ * @author Timothy Scott <timothy@auterion.com>
+ */
+
+
+/**
+ * Uart write period
+ *
+ * How long to wait, in Milliseconds, between writing actuator controls over Uart to the Roboclaw
+ * @unit ms
+ * @min 1
+ * @max 1000
+ * @group Roboclaw driver
+ */
+PARAM_DEFINE_INT32(RBCLW_WRITE_PER, 10);
+
+/**
+ * Encoder read period
+ *
+ * How long to wait, in Milliseconds, between reading wheel encoder values over Uart from the Roboclaw
+ * @unit ms
+ * @min 1
+ * @max 1000
+ * @group Roboclaw driver
+ */
+PARAM_DEFINE_INT32(RBCLW_READ_PER, 10);
```

Comment:
Same question here.

Reply:
The same applies for this: The Roboclaw doesn't provide any sort of interrupt or notification for when encoder data is available, so the driver just has to poll it periodically.

Reply:
I would hard code a reasonable value and not expose it as configurable unless actually necessary.